scripts/deploy/protocol/prepareScoutGameLaunchSafeTransaction.ts

```diff
@@ -411,37 +393,76 @@ task('prepareScoutGameLaunchSafeTransaction', 'Prepares a Safe transaction to la
       value: '0'
     };
 
-    await apiKit.estimateSafeTransaction(safeAddress, lockupCreateStreamTxData);
+    // await apiKit.estimateSafeTransaction(safeAddress, lockupCreateStreamTxData);
 
     safeTransactionData.push(lockupCreateStreamTxData);
 
     if (safeTransactionData.length === 0) {
       throw new Error('No valid transactions to propose');
     }
 
-    const safeTransaction = await protocolKitProposer.createTransaction({
-      transactions: safeTransactionData
-    });
-
-    log.info('Generated safe transaction input data');
-
-    const safeTxHash = await protocolKitProposer.getTransactionHash(safeTransaction);
-    const signature = await protocolKitProposer.signHash(safeTxHash);
-
-    const proposerAddress = privateKeyToAccount(PRIVATE_KEY).address;
+    // const safeTransaction = await protocolKitProposer.createTransaction({
+    //   transactions: safeTransactionData
+    // });
+
+    // log.info('Generated safe transaction input data');
+
+    // const safeTxHash = await protocolKitProposer.getTransactionHash(safeTransaction);
+    // const signature = await protocolKitProposer.signHash(safeTxHash);
+
+    // const proposerAddress = privateKeyToAccount(PRIVATE_KEY).address;
+
+    // Create and export transaction batch JSON
+    const exportTransactionBatch = async () => {
+      // Extract contract methods when possible
+      const transactionsWithMethods = safeTransactionData.map((tx) => {
+        // Basic transaction data
+        const txData = {
+          to: tx.to,
+          value: tx.value,
+          data: tx.data || null,
+          contractMethod: null,
+          contractInputsValues: null
+        };
+
+        // Try to extract method information if available
+        // This is a simplified approach - in a production environment you might want to
+        // decode the function call more precisely using the ABI
+        return txData;
+      });
+
+      const txBatchData = {
+        version: '1.0',
+        chainId: connector.chain.id.toString(),
+        createdAt: Date.now(),
+        meta: {
+          txBuilderVersion: '1.17.1'
+        },
+        transactions: transactionsWithMethods
+      };
+
+      // Write to file
+      const filePath = path.join(process.cwd(), 'scout-protocol-tx-batch.json');
+
+      fs.writeFileSync(filePath, JSON.stringify(txBatchData, null, 2));
+
+      log.info(`Transaction batch exported to ${filePath}`);
+      return filePath;
+    };
 
-    log.info(`Proposing transaction to safe with hash ${safeTxHash}`);
+    const exportedFilePath = await exportTransactionBatch();
+    log.info(`Exported transaction batch to ${exportedFilePath}`);
 
-    // Propose transaction to the service
-    await apiKit.proposeTransaction({
-      safeAddress,
-      safeTransactionData: safeTransaction.data,
-      safeTxHash,
-      senderAddress: proposerAddress,
-      senderSignature: signature.data
-    });
+    // // Propose transaction to the service
```

Comment:
instead of commenting code, can you add a flag at the top, like "USE_API" or "EXPORT_ONLY"

Reply:
yeah good point